packages/eslint-plugin/tests/rules/consistent-type-assertions.test.ts

```diff
@@ -195,6 +195,12 @@ ruleTester.run('consistent-type-assertions', rule, {
           line: 6,
         },
       ],
+      output: `
+const x = new Generic<int>() as Foo;
+const x = b as A;
+const x = [1] as readonly number[];
+const x = 'string' as a | b;
+const x = { key: 'value' } as const;`,
```

Comment:
Autofixers have a bad habit of changing PEMDAS accidentally. We'll need to make sure this one is safe for a few edge cases:

* `<A | B>('string');`
* `<A>!'string';`
* `<A>a + b;`
* `<Foo>(new Generic<string>());`
* `(new (<Foo>Generic<string>)());`

Related PRs:
* https://github.com/typescript-eslint/typescript-eslint/pull/5997
* https://github.com/typescript-eslint/typescript-eslint/pull/6118
* https://github.com/typescript-eslint/typescript-eslint/pull/6569

Are there any others you can think of?